rplugin/node/nvim_typescript/src/index.ts

```diff
@@ -503,11 +503,20 @@ export default class TSHost {
   }
 
   async openBufferOrWindow(file: string, lineNumber: number, offset: number) {
+
+    const fileIsAlreadyFocused =
+        await this.getCurrentFile().then(currentFile => file === currentFile);
+
+    if (fileIsAlreadyFocused) {
+        await this.nvim.command(`call cursor(${lineNumber}, ${offset})`);
+        return;
+    }
+
     const windowNumber = await this.nvim.call('bufwinnr', file);
     if (windowNumber != -1) {
       await this.nvim.command(`${windowNumber}wincmd w`);
     } else {
-      await this.nvim.command(`e! ${file}`);
```

Comment:
I guess that with the `:e!`, if you were editing a file with unsaved changes _and_ `:TSDef`'d something in this same file, you would lose your changes :(